src/rgb21/issuer.rs

```diff
@@ -0,0 +1,203 @@
+use std::str::FromStr;
+
+use amplify::confinement::SmallBlob;
+use amplify::Wrapper;
+use bp::dbc::Method;
+use rgbstd::containers::ValidContract;
+use rgbstd::interface::{BuilderError, ContractBuilder, IfaceClass, TxOutpoint};
+use rgbstd::invoice::{Allocation, Precision};
+use rgbstd::stl::{AssetSpec, Attachment, ContractTerms, MediaType, RicardianContract};
+use rgbstd::{AltLayer1, AssetTag, GenesisSeal, Identity, TokenIndex};
+use strict_encoding::InvalidRString;
+
+use super::{EmbeddedMedia, Rgb21, TokenData};
+use crate::{IssuerWrapper, SchemaIssuer};
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug, Display, Error)]
+#[display(doc_comments)]
+pub enum IssuerError {
+    /// contract genesis doesn't support allocating to liquid seals; request
+    /// liquid support first.
+    NoLiquidSupport,
+    /// the amount of token fractions in outputs exceeds 1.
+    FractionOverflow,
+    /// attachment has a type which is not allowed for the token
+    InvalidAttachmentType,
+    /// allocation of unknown token ID {0}
+    UnknownToken,
```

Comment:
these seem unused so please remove them or map them 

moreover the last ones are missing a dot at the end of the message

Reply:
I think maybe we should keep these items, and the reasons:
1. for `FractionOverflow `,  maybe we can perform the token owned fraction algorithm operation to check if the owned fraction is over flow.
2. for `InvalidAttachmentType `, maybe we can add attach type into the global types?
3. for `UnknownToken`, we can check if or not the token id is exist.

Reply:
I'm not sure we should do these checks, but I'm pretty sure that we should not add unused error variants. If you'll implement the mentioned checks then I'll review them, otherwise please drop these